controllers/suite_test.go

```diff
@@ -70,11 +74,33 @@ var _ = BeforeSuite(func(done Done) {
 	Expect(err).ToNot(HaveOccurred())
 	Expect(k8sClient).ToNot(BeNil())
 
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	err = (&ServiceReconciler{
+		Client:              k8sManager.GetClient(),
+		BackendController:   &BackendController{},
+		Recorder:            k8sManager.GetEventRecorderFor("autoneg-controller"),
+		Log:                 ctrl.Log.WithName("controllers").WithName("Service"),
+		ServiceNameTemplate: serviceNameTemplate,
+		AllowServiceName:    true,
+	}).SetupWithManager(k8sManager)
+	Expect(err).ToNot(HaveOccurred())
+
+	go func() {
+		defer GinkgoRecover()
+		err = k8sManager.Start(ctrl.SetupSignalHandler())
+		Expect(err).ToNot(HaveOccurred())
+	}()
+
 	close(done)
 }, 60)
 
 var _ = AfterSuite(func() {
 	By("tearing down the test environment")
+	gexec.KillAndWait(5 * time.Second)
```

Comment:
i'm no expert on both go and autoneg but 2 general comments:

i've noticed often our integration tests are failing on timeouts, so i'm a bit scared to see a timeout added in the test. Are we sure we can't implement this without a timeout? or set the timeout very high?

also, `gexec.KillAndWait` sounds very dramatic if this translates to a unix call, shoudn't we stop first?